pkg/adapter/v2/cloudevents.go

```diff
@@ -302,13 +302,21 @@ func (c *client) reportMetrics(ctx context.Context, event cloudevents.Event, res
 	if c.reporter == nil {
 		return
 	}
+	var scheme string
+	parsedUrl , err := url.Parse(event.Source())
+	if err != nil {
+		scheme = "unknown"
+	}
+	scheme = parsedUrl.Scheme
+	
```

Comment:
```suggestion
```

I'm not super familiar with the metrics part of the TLS feature track, but IIRC there isn't any need to change the metrics reporting in the adapterv2. cc @pierDipi for confirmation here


Also, as explained in my other comment, looking at the scheme of the event source isn't guaranteed to tell you whether this event was delivered with http or https.

Reply:
We need to change this part so that we know the outbound requests using https vs http, however, the value of the scheme tag needs to come from the HTTP URL that we get from `cfg.Env.GetSink()` in `NewClient()`